docs/concept-workflows.md

```diff
@@ -60,18 +74,34 @@ public class SubscriptionWorkflowImpl implements SubscriptionWorkflow {
 }
 ```
 
-Again, note that this code directly implements the business logic. If any of the invoked operations (aka Activities) takes a long time, the code is not going to change. It is okay to block on `chargeMonthlyFee` for a day if the downstream processing service is down that long. The same way that blocking sleep for 30 days is a normal operation inside the Workflow code.
+Again, note that this code directly implements the business logic: 
 
-Temporal has practically no scalability limits on the number of open Workflow instances. So even if your site has hundreds of millions of consumers, the above code is not going to change.
+- If any of the invoked operations (aka Activities) takes a long time, the code is not going to change. 
+- It is okay to block on `chargeMonthlyFee` for a day if the downstream processing service is down that long.
+- In the same way, blocking the Workflow with `sleep` for 30 days is a routine operation.
+- Temporal has practically no scalability limits on the number of concurrently open Workflow instances. So even if your site has hundreds of millions of consumers, the above code is not going to change.
 
-The commonly asked question by developers that learn Temporal is "How do I handle Workflow worker process failure/restart in my Workflow"? The answer is that you do not. **The Workflow code is completely oblivious to any failures and downtime of workers or even the Temporal service itself**. As soon as they are recovered and the Workflow needs to handle some event, like timer or an Activity completion, the current state of the Workflow is fully restored and the execution is continued. The only reason for a Workflow failure is the Workflow business code throwing an exception, not underlying infrastructure outages.
+### Frequently asked questions
 
-Another commonly asked question is whether a worker can handle more Workflow instances than its cache size or number of threads it can support. The answer is that a Workflow, when in a blocked state, can be safely removed from a worker.
-Later it can be resurrected on a different or the same worker when the need (in the form of an external event) arises. So a single worker can handle millions of open Workflow executions, assuming it can handle the update rate.
+- **Can I write infinitely long running Workflows?**
+    - Workflows that are intended to run indefinitely should be written with some care. Temporal stores the complete event history for the entire life of a Workflow Execution, and there is a maximum size enforced by the service.
+    - The idiomatic way to handle this is to return `ContinueAsNew` (available in all SDKs) at a reasonable cutoff point to complete the current Workflow Execution and start a new execution with the same Workflow Id, passing it the appropriate parameters to continue.
```

Comment:
THIS is new content - inserted to directly address a very popular usecase that is not well addressed by docs (as far as I can tell)

i wanted to directly link ContinueAsNew in each SDK, but discovered that the Java SDK does not have a doc for `ContinueAsNew`

Reply:
Ryland's tweak to the title, and a couple more notes from Cully and Ryland

```suggestion
- **Is there a limit to how long workflows can run?**
    - Workflows intended to run indefinitely should be written with some care. Temporal stores the complete event history for the entire life of a Workflow Execution, and there is a maximum size of 50,000 events enforced by the service.
    - You should not get close to this limit; Temporal warns every 10,000 events.
    - The idiomatic way to handle indefinitely long workflows is to return `ContinueAsNew` (available in all SDKs) at a reasonable cutoff point to complete the current Workflow Execution and start a new execution with the same Workflow Id, passing it the appropriate parameters to continue.
    - If you are using [Signals](https://docs.temporal.io/docs/go-signals/), you should make sure to do an async drain on the Signal Channel or they will be dropped.
```